src/read.jl

```diff
@@ -197,30 +197,30 @@ function read(io::IO, ::Type{Function})
 end
 
 # read Number, String, Nullable, Bool
-read(io::IO, ::Type{T}) where {T <: Integer} = Parsers.parse(io, T)
-read(io::IO, ::Type{T}) where {T <: AbstractFloat} = Parsers.parse(io, T)
+read(io::IO, ::Type{T}; kwargs...) where {T <: Integer} = Parsers.parse(io, T)
+read(io::IO, ::Type{T}; kwargs...) where {T <: AbstractFloat} = Parsers.parse(io, T)
 
-function read(io::IO, T::Type{Char})
+function read(io::IO, T::Type{Char}; kwargs...)
     @expect '"'
     c = Char(readbyte(io))
     @expect '"'
     return c
 end
-read(io::IO, ::Type{Dates.Date}, format=Dates.ISODateFormat) = Dates.Date(read(io, String), format)
-read(io::IO, ::Type{Dates.DateTime}, format=Dates.ISODateFormat) = Dates.DateTime(read(io, String), format)
-read(io::IO, ::Type{T}) where {T <: Enum} = Core.eval(parentmodule(T), read(io, Symbol))
+read(io::IO, ::Type{Date}; dateformat=ISODateFormat, kwargs...) = Date(read(io, String; kwargs...), dateformat)
+read(io::IO, ::Type{DateTime}; datetimeformat=ISODateTimeFormat, kwargs...) = DateTime(read(io, String; kwargs...), datetimeformat)
```

Comment:
I'd rather call this keyword argument just `dateformat=ISODateTimeFormat`; it's pretty standard in the Dates module and elsewhere to just always call this thing `dateformat` because it's not really specific to Date or DateTime.